packages/node/src/indexer/dictionary/v1/substrateDictionaryV1.ts

```diff
@@ -62,7 +65,7 @@ function getBaseHandlerKind<
   ds: SubstrateDataSource,
   handler: SubstrateHandler,
   getDsProcessor: (ds: SubstrateDatasource) => P,
-): SubstrateHandlerKind {
+): SubstrateHandlerKind | undefined {
```

Comment:
I think this could throw instead of being undefined. Then if we added another type we would be notified about the failure rather than silently not handling code.

Alternatively the switch that consumes the result of this function could also throw